src/OauthTokenFileStorage.php

```diff
@@ -200,7 +200,14 @@ public function checkRequirements(): void {
    * {@inheritdoc}
    */
   public function removeToken(): void {
-    $this->saveToken([]);
+    // Prevents a token file from being created if it doesn't already exist.
+    if (file_exists($this->tokenFilePath)) {
```

Comment:
It is not enough to call `file_exist()`, you must run the same sanity checks as you can see in `removeTokenFile()` otherwise calling this function with a insufficiently configured private filesystem can generate errors. This the reason why I tried to call `saveToken([])` here simple and eliminate calling this method in `apigee_edge_cache_flush()` when it should not be call.

Reply:
Isn't this "fix" got required **only** because you had added this call to the authentication form?
https://github.com/apigee/apigee-edge-drupal/pull/103/files#diff-c1ab508d8adb4da185ab3867ced29eb6R364

Reply:
@Jaesin Please investigate this one too.

Reply:
I dont' think there is anything to fix here. `::saveToken` handles validation.

Reply:
Pulled this patch and run an updb commend, I could run a "drush cr" command too on a site that private filesystem has not been configured. I got the error that I referred to in my first comment. `saveToken()` may handles this issue but `file_exists()` triggers an error if you call it in an non-registered file stream wrapper.

```
$ drush updb -y 
 --------------- -------------------------- --------------- --------------------------------------------------------------------------------- 
  Module          Update ID                  Type            Description                                                                      
 --------------- -------------------------- --------------- --------------------------------------------------------------------------------- 
  admin_toolbar   8001                       hook_update_n   Rebuild routes to mitigate issue 2938884.   @see                                 
                                                             https:www.drupal.orgprojectadmin_toolbarissues2938884                            
  node            node_revision_views_data   post-update     Clear caches due to updated views data.                                          
  system          states_clear_cache         post-update     Force cache clear to ensure aggregated JavaScript files are regenerated.   @see  
                                                             https:www.drupal.orgprojectdrupalissues2995570                                   
 --------------- -------------------------- --------------- --------------------------------------------------------------------------------- 

 // Do you wish to run the specified pending updates?: yes.                                                             

 [notice] Update started: admin_toolbar_update_8001
 [error]  The "company" entity type does not exist. 
 [error]  Update failed: admin_toolbar_update_8001 
 [warning] file_exists(): Unable to find the wrapper "private" - did you forget to enable it when you configured PHP? OauthTokenFileStorage.php:204
 [warning] file_exists(): Unable to find the wrapper "private" - did you forget to enable it when you configured PHP? OauthTokenFileStorage.php:204
```

Reply:
@cnovak Should I work on this one or leave it for a follow up task?

Reply:
Let's add this to the GitHub issue queue and tag it as a known issue